app/src/main/java/sk/inloop/support/v4/app/FragmentStatePagerAdapter.java

```diff
@@ -0,0 +1,248 @@
+package sk.inloop.support.v4.app;
+
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v4.view.PagerAdapter;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by adammihalik on 05/12/2016.
+ */
+
+public abstract class FragmentStatePagerAdapter extends PagerAdapter {
+
+    private static final String TAG = FragmentStatePagerAdapter.class.getSimpleName();
+    private static final boolean DEBUG = false;
+
+    private final FragmentManager mFragmentManager;
+    private FragmentTransaction mCurTransaction = null;
+
+    private List<Fragment.SavedState> mSavedState;
+    private List<Fragment> mFragments;
+    private Fragment mCurrentPrimaryItem = null;
+
+    public FragmentStatePagerAdapter(FragmentManager fm) {
+        mFragmentManager = fm;
+        mFragments = buildFragmentsList();
+        mSavedState = buildFragmentStatesList();
+    }
+
+    /**
+     * Return not null and empty {@link List} for storing of {@link android.support.v4.app.Fragment.SavedState}s. Depend on adapter's purpose (only-read, only-insert, insert-delete), developer have to decide which implementation if list should be used. Default is {@link ArrayList}.
+     */
+    @NonNull
+    protected List<Fragment.SavedState> buildFragmentStatesList() {
+        return new ArrayList<Fragment.SavedState>();
+    }
+
+    /**
+     * Return not null and empty {@link List} for storing of initialized {@link Fragment}s. Depend on adapter's purpose (only-read, only-insert, insert-delete), developer have to decide which implementation if list should be used. Default is {@link ArrayList}.
+     */
+    @NonNull
+    protected List<Fragment> buildFragmentsList() {
```

Comment:
Why is this a protected method? Is there any reason why you would override it? (I don't think so)